### PR TITLE
Expand Functions introductory notebook

### LINKED ISSUE #118: Functions: provide intermediate exercises

### PR DESCRIPTION
**Pull request by edoardob90**
Fixes #118

### CODE REVIEW

File: functions.ipynb [hunk 1]
```diff
@@ -402,6 +412,104 @@
...
+    "Write a Python function called `calculate_area` that takes three parameters: `length` (a float), `width` (a float), and `unit` (a string with a **default** value of `\"cm\"`).\n",
+    "The function should calculate the area of a rectangle based on the given length and width, and return the result **as a tuple** with the correct, default unit (i.e., `cm^2`).\n",
+    "If the unit parameter is \"m\", the function should convert the length and width from meters to centimeters before calculating the area.\n",
+    "\n",
+    "Your solution function **must** handle the following input units (the output unit is **always** `cm^2`):\n",
+    "\n",
+    "- centimeters (`cm`)\n",
+    "- meters (`m`)\n",
+    "- millimeters (`mm`)\n",
+    "- yards (`yd`)\n",
```
Comment (despadam) [inline functions.ipynb:481]: Should we provide the values for yards and feet to do the conversion?

**Review by despadam — CHANGES_REQUESTED** [review]
Great job, only left comments about a few details.

File: tutorial/tests/test_functions.py [hunk 2]
```diff
@@ -1,22 +1,169 @@
...
 
 #
-# Exercise 1: Longest Sequence
+# Exercise 1: a `greet` function
+#
+
+
+def reference_greet(name: str, age: int) -> str:
+    """Reference solution for the greet exercise"""
+    return f"Hello {name}, you are {age} years old."
```
Comment (edoardob90) [inline tutorial/tests/test_functions.py:22]:
```suggestion
    return f"Hello {name}! You are {age} years old."
```

File: functions.ipynb (hunk 1 above)
Comment (edoardob90) [reply functions.ipynb:481]: Mmm, I would say "no". It's fairly easy to look it up online. If you feel it's too much (😄), then feel free to add the conversion factors to the problem's text.

**Comment by edoardob90** [comment]
@despadam Thanks for the comments! I addressed them and fixed a few conflicts that came up. Have a second look when you've some time.

File: functions.ipynb [hunk 3]
```diff
@@ -405,6 +415,107 @@
...
+    "- feet (`ft`)\n",
+    "\n",
+    "If you pass an unsupported unit, the function should **return** a string with the error message: `Invalid unit: <unit>`, where `<unit>` is the unsupported unit.\n",
+    "\n",
+    "<div class=\"alert alert-block alert-warning\">\n",
+    "    <h4><b>Note</b></h4>\n",
+    "    <strong>Do not</strong> forget to write a proper docstring and add the correct type hints to the parameters and the return value.\n",
+    "</div>"
```
Comment (despadam) [inline functions.ipynb:492]:
```suggestion
    "</div>"
    <div class="alert alert-block alert-info">
    <h4><b>Hints</b></h4>
    <ul>
        <li>1 yd = 91.44 cm</li>
        <li>1 ft = 30.48 cm</li>
    </ul>
</div>
```

**Review by despadam — CHANGES_REQUESTED** [review]

File: functions.ipynb (hunk 1 above)
Comment (despadam) [reply functions.ipynb:484]: Please see my requested changes below - I added a hint

File: functions.ipynb (hunk 3 above)
Comment (edoardob90) [reply functions.ipynb:492]: Just be careful when editing HTML inside notebooks: forgetting or misplacing a tag breaks everything.

Comment (despadam) [reply functions.ipynb:492]: You are right, I forgot the double quotes 😅

**Review by despadam — APPROVED** [review]

**Pull request merged**